src/bin/playground.cpp

```diff
@@ -1,10 +1,436 @@
+      return;
+    case CompressedVectorType::FixedWidthInteger2Byte:
+      export_values(ofstream, dynamic_cast<const FixedWidthIntegerVector<uint16_t>&>(compressed_vector).data());
+      return;
+    case CompressedVectorType::FixedWidthInteger1Byte:
+      export_values(ofstream, dynamic_cast<const FixedWidthIntegerVector<uint8_t>&>(compressed_vector).data());
+      return;
+    case CompressedVectorType::BitPacking:
+      export_compact_vector(ofstream, dynamic_cast<const BitPackingVector&>(compressed_vector).data());
+      return;
+    default:
+    Fail("Any other type should have been caught before.");
+  }
+}
+
+void write_dict_segment_to_disk(const std::shared_ptr<DictionarySegment<int>> segment, const std::string& filename) {
+  /*
+   * Write dict segment to given file using the following format:
+   * 1. Number of elements in dictionary
+   * 2. Number of elements in attribute_vector
+   * 3. AttributeVectorCompressionID aka. size of int used in attribute vector
+   * 3. Dictionary values
+   * 4. Attribute_vector values
+   *
+   * For this exercise we assume an <int>-DictionarySegment with an FixedWidthIntegerVector<uint16_t> attribute_vector.
+   * As a next step we should use the AttributeVectorCompressionID to define the type of the FixedWidthIntegerVector
+   * and perhaps also write out the type of the DictionarySegment.
+   */
+  std::ofstream chunk_file;
+  chunk_file.open(filename, std::ios::out | std::ios::binary | std::ios::app);
+
+  // We will later mmap to an uint32_t vector/array. Therefore, we store all metadata points as uint32_t.
+  // This wastes up to three bytes of compression per metadata point but makes mapping much easier.
+  export_value(chunk_file, static_cast<uint32_t>(segment->dictionary()->size()));
+  export_value(chunk_file, static_cast<uint32_t>(segment->attribute_vector()->size()));
+
+  const auto compressed_vector_type_id = static_cast<uint32_t>(infer_compressed_vector_type_id<int>(*segment));
+  export_value(chunk_file, compressed_vector_type_id);
+
+  export_values<int32_t>(chunk_file, *segment->dictionary());
+
+  export_compressed_vector(chunk_file, *segment->compressed_vector_type(),
+                          *segment->attribute_vector());
+  chunk_file.close();
+}
+
+std::vector<uint32_t> generate_segment_offset_ends(const std::shared_ptr<Chunk> chunk) {
+  const auto segment_count = chunk->column_count();
+  auto segment_offset_ends = std::vector<uint32_t>(segment_count);
+  // Chunk Header Size
+  auto offset_end = static_cast<uint32_t>(4 + segment_count * 4);
+
+  for (auto segment_index = size_t{0}; segment_index < segment_count; ++segment_index) {
+    // 4 Byte Dictionary Size + 4 Byte Attribute Vector Size + 4 Compressed Vector Type ID
+    offset_end += 12;
+
+    const auto abstract_segment = chunk->get_segment(static_cast<ColumnID>(static_cast<uint16_t>(segment_index)));
```

Comment:
Why not use `DictionarySegment<T>::memory_usage`?

Reply:
Because I did not know about it until now 😄 I will have a look into it.

Reply:
Or as a general OOP design, I think it would be nice to give the Chunk the ability to calculate and return its memory usage. This would additionally help to avoid errors imo.
There already is a function `memory_usage` defined in `chunk.hpp` which is currently marked as an estimate for debugging purposes. I think it would be interesting to find out why its marked as an estimate (the corresponding `memory_usage` function on `DictionarySegment` isn't) and if it would be possible to extend it to make it a safe calculation.

Reply:
```/**
   * For debugging purposes, makes an estimation about the memory used by this chunk and its segments
   */
  size_t memory_usage(const MemoryUsageCalculationMode mode) const;
  
```

Reply:
From my sporadic tests I found, that the `memory_usage` method returns exactly the same value as our calculation.